### ISSUE TITLE
Blob type failing to save in Safari 10.1.2

### ISSUE THREAD

**Issue by alex-milburn** [gilf/angular2-indexeddb#41]
Blob is failing to save using angular2-indexeddb.

Code sample:

`     // create blob:
      const aFileParts = ['<a id="a"><b id="b">foo!</b></a>'];
      const oMyBlob = new Blob(aFileParts, {type : 'text/html'});

      console.log('blob type' + oMyBlob.type); // outputs as 'text/html'

      // initialize my indexeddb store:
      return this.initializeStores().then(() => {

        // add 'oMyBlob' to the FileData data store:
        return this.db.add('FileData',
          { FileName: 'foo', FileData: oMyBlob, FileType: 'audio' }).then(() => {

            // Success
            console.log('added ' + 'foo' + ' to FileData store.');

            // Get the file from the FileData store
            return this.db.getByIndex('FileData', 'FileName', 'foo').then((record) => {
              return Promise.resolve();
            });

          }, (error) => {
            console.log(error);
            this.handleError(error)
            return Promise.reject(error);
          });
      }, (error) => {
        this.handleError(error);
        return Promise.reject(error);
      });`

When i look inside the store the field for the blob is 'null'. This works when i use indexeddb without the angular module.

Is this a known issue? i like using this module, but this issue is a deal breaker for me. I'm hoping there is a workaround.

**Comment by gilf** [comment]
Hi,
Is it something that happen on Mobile Safari? 
There is a known problem with Mobile Safari, Blobs and IndexedDB - [See this Stack Overflow question](https://stackoverflow.com/questions/40393488/mobile-safari-10-indexeddb-blobs)

**Comment by alex-milburn** [comment]
Hey Gilf, thanks for replying. I managed to get around this by using ArrayBuffers (not ideal as i have to do some extra processing). I managed to get blobs working on Desktop Safari (turns out there was a problem with the index key path that i was using).

However, on Mobile Safari i ran into the 'unknown error' when trying to save the blob to indexeddb.

Something to be aware of - in the guide you use 'id' as the index key path (although it's not an actual property on the store). This was causing me issues as i had just blindly followed your guide. It wasn't until later that i found my indexeddb had issues due to this (mostly my fault to be honest).

Alex.
